src/main/java/com/aws/greengrass/status/FleetStatusService.java

```diff
@@ -295,24 +296,40 @@ public void schedulePeriodicFleetStatusDataUpdate(boolean isDuringConnectionResu
                 initialDelay, periodicPublishIntervalSec, TimeUnit.SECONDS);
     }
 
+    private AtomicReference<OverallStatus> getOverAllStatus() {
+        Instant now = Instant.now();
+        AtomicReference<OverallStatus> overAllStatus = new AtomicReference<>(OverallStatus.HEALTHY);
+        this.kernel.orderedDependencies().forEach(service -> {
+            serviceFssTracksMap.put(service, now);
+            overAllStatus.set(getOverallStatusBasedOnServiceState(overAllStatus.get(), service));
+        });
+        return overAllStatus;
+    }
+
     @SuppressWarnings("PMD.UnusedFormalParameter")
     private void handleServiceStateChange(GreengrassService greengrassService, State oldState,
                                           State newState) {
+
+        Set<GreengrassService> erroredComponentSet = new HashSet<>();
```

Comment:
Why do you need a copy of the `erroredComponentsSet`? is it not possible to reuse the existing set?

Reply:
Also, it looks like you are adding instances of services which recovered in the `erroredComponentSet`. Maybe, rename with a more relevant name? like `recoveredComponentSet`?

Reply:
Updated it to one liner